### PR TITLE
Add .gitattribute assisted language detection to blame, diff and render

### PR DESCRIPTION
**Pull request by zeripath**
Use check attribute code to check the assigned language of a file and send that in to
chroma as a hint for the language of the file.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by jolheiser** [comment]
Looks like some `gogit` variants need changes. 

Also, could you maybe add some docs for this? Given the `app.ini` version is already undocumented, this seems a bit hidden. 🙁

**Comment by silverwind** [comment]
Should probably support [`gitlab-language`](https://docs.gitlab.com/ee/user/project/highlighting.html) too.

**Comment by zeripath** [comment]
@silverwind done
@jolheiser done

File: modules/git/repo_language_stats_gogit.go [hunk 1]
```diff
@@ -113,6 +104,16 @@ func (repo *Repository) GetLanguageStats(commitID string) (map[string]int64, err
 
 					sizes[language] += f.Size
 
+					return nil
+				} else if language, has := attrs["gitlab-language"]; has && language != "unspecified" && language != "" {
```
Comment (lafriks) [inline modules/git/repo_language_stats_gogit.go:109]: Would be nice to handle also to strip everything after and including `?` character:
attels

Comment (zeripath) [reply modules/git/repo_language_stats_gogit.go:109]: done

**Review by lafriks — APPROVED** [review]

**Review by jolheiser — APPROVED** [review]

**Comment by lafriks** [comment]
🚀

**Pull request merged**